pgrx/src/datum/numeric.rs

```diff
@@ -40,30 +40,12 @@ pub struct Numeric<const P: u32, const S: u32>(pub(crate) AnyNumeric);
 /// to represent any Rust primitive value from `i128::MIN` to `u128::MAX` and anything in between.
 ///
 /// Generally, this is the type you'll want to use as function arguments.
+///
+#[derive(Debug, Clone)]
 pub struct AnyNumeric {
-    pub(crate) inner: pg_sys::Numeric,
-    pub(crate) need_pfree: bool,
-}
-
-impl Clone for AnyNumeric {
-    /// Performs a deep clone of this [`AnyNumeric`] into the [`pg_sys::CurrentMemoryContext`].
-    fn clone(&self) -> Self {
-        unsafe {
-            let copy = PgMemoryContexts::CurrentMemoryContext
-                .copy_ptr_into(self.inner, varsize(self.inner.cast()));
-            AnyNumeric { inner: copy, need_pfree: true }
-        }
-    }
-}
-
-impl Drop for AnyNumeric {
-    fn drop(&mut self) {
-        if self.need_pfree {
-            unsafe {
-                pg_sys::pfree(self.inner.cast());
-            }
-        }
-    }
+    // we represent a NUMERIC as opaque bytes -- we never inspect these ourselves, only a pointer
+    // to them (cast as a [`Datum`]) are passed to Postgres
+    pub(crate) inner: Vec<u8>,
```

Comment:
This never grows, so `Box<[u8]>` would be slightly more efficient

Reply:
How does one make a `[u8]` from arbitrary bytes?

Reply:
you can use `.into()` to make a `Vec<u8>` into a `Box<[u8]>`

Reply:
hmm.  Okay.  Down here (https://github.com/pgcentralfoundation/pgrx/pull/1216#discussion_r1265746694), is there a more direct way to do that?